pkg/controller/stacks/stack/stack.go

```diff
@@ -160,29 +188,169 @@ func (h *stackHandler) update(ctx context.Context) (reconcile.Result, error) {
 	return reconcile.Result{}, nil
 }
 
-func (h *stackHandler) createNamespacedRoleBinding(ctx context.Context, owner metav1.OwnerReference) error {
-	cr := &rbac.Role{
+// createPersonaClusterRoles creates admin, edit, and view clusterroles that are
+// namespace+stack+version specific
+func (h *stackHandler) createPersonaClusterRoles(ctx context.Context, labels map[string]string) error {
```

Comment:
this function is a pretty clean approach for creating the personal cluster roles that include all the CRDs for the stack and the correct aggregation labels etc., but it could use a few comments sprinkled through to explain the steps being taken and the context around why they are taken. that will definitely be helpful for future maintainers.

Same with the `generateNamespaceClusterRoles` function too.